src/core/resolver/dns/c_ares/grpc_ares_wrapper.cc

```diff
@@ -1179,12 +1180,10 @@ static void grpc_cancel_ares_request_impl(grpc_ares_request* r) {
 
 void (*grpc_cancel_ares_request)(grpc_ares_request* r) =
     grpc_cancel_ares_request_impl;
+grpc_core::Mutex ares_setup_mu_;
```

Comment:
I'm not sure adding a mutex in gRPC is the right thing to do. If an application doesn't use curl and gRPC together, it would not see this race and it doesn't need this mutex. I think application needs to make sure that `ares_library_init()` and `ares_library_cleanup()` are not called concurrently from different threads.

Reply:
I added the mutex to address the previous comment (`Calling them may cause race conditions when other parts of the binary calls these functions concurrently.`). Ares is not intended to be thread-safe (https://github.com/c-ares/c-ares/issues/449#issuecomment-996681949), so I think the previous comment make sense and we need to handle it here.

Reply:
Again I think it's the application's job to make sure it's not causing this race condition. Adding the mutex here is not necessary for users who do not use curl and gRPC in the same binary. In general, we don't want to add things in gRPC that would only address a specific use case or problem. This solution would actually add mutex ops to all gRPC users which is not necessary.

Reply:
I see. Actually adding mutex here also doesn't help when user use curl and gRPC in the same binary, it only help synchronize `ares_library_init()` in the same process (i.e. when gRPC call the `ares_library_init` in different thread). I saw that the previous comment noted their concern about it and tried to address it, but it seems it's not a problem in gRPC right now so it shouldn't be a problem to remove this mutex.

Removed.